src/execute.js

```diff
@@ -17,7 +19,13 @@ const executeHttpRequest = (input, options) => {
   return input.z
     .request(options)
     .then(throwForStatus)
-    .then(resp => createJSONtool().parse(resp.content));
+    .then(resp => {
+      try {
+        return createJSONtool().parse(resp.content);
+      } catch (err) {
+        return querystring.parse(resp.content);
```

Comment:
OAuth1 spec says that the response body returned by `getRequestToken` and `getAccessToken` should be URL encoded. For example, `oauth_token=abcd&oauth_token_secret=1234`. So we need to parse it properly here.

Reply:
Does this affect existing apps, as we'll now try a new parsing that we didn't before? Also, we won't throw an error for not being able to parse JSON content.

Wondering if we can find a different spot to put this processing to avoid a breaking change.

Reply:
This is a good point! Made a change in fa98ad7 so we're parsing form body only when it's oauth1.

Reply:
On that re-encoding from form body to JSON, does that only happen for Request Schemas and not for performs that are functions? It'd be super confusing to change up the response type on the dev if they are also going to manipulate the response.

Even if it's only for request schema's, what impact does this have on logging? Will we log JSON when in fact we got form body?

Reply:
The re-encoding is also going to happen for perform functions, which can be really confusing. I totally agree. Pushed da7d824, where form body parsing only happens for `getRequestAccess` and `getAccessToken` and only when auth type is `oauth1`. Would like to know if I miss anything. Thanks!